src/server/main_service.cc

```diff
@@ -2186,7 +2209,9 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
       ServerState::tlocal()->exec_freq_count[descr]++;
     }
 
-    if (absl::GetFlag(FLAGS_multi_exec_squash) && state == ExecEvalState::NONE) {
+    if (absl::GetFlag(FLAGS_multi_exec_squash) && state == ExecEvalState::NONE &&
+        !cntx->conn_state.exec_info.HasClientCommand() &&
```

Comment:
@dranikpg if there is a client command within the `multi/exec` block or if `tracking is on` then we don't squash, we fall back to normal dispatch. 

The two changes introduced bring us back to full compat realm with redis/valkey.

Basically redis allow: 

```
MULTI 
CLIENT TRACKING ON 
EXEC
GET FOO <---- Now is being tracked
```

These are the only two changes. You can pick them from the last commit. The first commit is basically the previous PR introduced.

Also note that the `sentinel` tests failed because I disabled `CLIENT` commands in `MULTI/EXEC` block. This is no longer the case, but we disable squashing for all client tracking cases.